beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2900,7 +2902,7 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                 snapshot_cache.insert(
                     BeaconSnapshot {
                         beacon_state: state,
-                        beacon_block: signed_block,
+                        beacon_block: signed_block.clone(),
```

Comment:
I don't think we need this clone

```suggestion
                        beacon_block: signed_block,
```